amespahdbpythonsuite/transitions.py

```diff
@@ -243,33 +312,72 @@ def calculatedtemperature(self, e: float, **keywords) -> None:
         i = 0
 
         nuids = len(self.uids)
+        self._atoms()
 
         for uid in self.uids:
             # Start timer.
             tstart = time.perf_counter()
+            # Instatiate model energy and temperature dictionaries.
+            self.model['energy'][uid] = {}
 
             print("SPECIES                          : %d/%d" % (i + 1, nuids))
             print("UID                              : %d" % uid)
+
+            if (
+                keywords.get('approximate')
+                or keywords.get('star')
+                or keywords.get('isrf')
+            ):
+                global charge
+                global nc
+
+                charge = self.pahdb["species"][uid]["charge"]
+                nc = self.atoms[uid]["nc"]
+
+            if keywords.get('star') or keywords.get('isrf'):
+                energy = Transitions.mean_energy(**keywords)
+
+                if not isinstance(energy, float):
+                    raise TypeError(
+                        'Expecting temperature as float type or isrf keyword'
+                    )
+
+                self.model['energy'][uid]['sigma'] = np.sqrt(
+                    Transitions.mean_energy_squared(**keywords) - energy**2
+                )
+
+            else:
+                energy = e
+                self.model['energy'][uid]['sigma'] = 0.0
+
+            self.model['energy'][uid]['e'] = energy
+
             print(
-                "MEAN ABSORBED ENERGY             : %f +/- %f eV"
-                % (e / 1.6021765e-12, 0.0)
+                "MEAN ABSORBED ENERGY             :"
+                f" {self.model['energy'][uid]['e'] / 1.6021765e-12} +/-"
+                f" {self.model['energy'][uid]['sigma'] / 1.6021765e-12} eV"
             )
 
             global frequencies
             frequencies = np.array([d["frequency"] for d in self.data[uid]])
 
-            Tmax = optimize.brentq(self.attainedtemperature, 2.73, 5000.0)
+            if keywords.get('approximate'):
+                Tmax = optimize.brentq(
+                    Transitions.approximate_attained_temperature, 2.73, 5000.0
+                )
+            else:
+                Tmax = optimize.brentq(Transitions.attained_temperature, 2.73, 5000.0)
 
-            print("MAXIMUM ATTAINED TEMPERATURE     : %f Kelvin" % Tmax)
+            self.model["temperatures"][uid] = Tmax
 
-            self.model["temperatures"].append({"uid": uid, "temperature": Tmax})
+            print("MAXIMUM ATTAINED TEMPERATURE     : %f Kelvin" % Tmax)
```

Comment:
Use `message` here.

Reply:
This is in the loop for each uid, so calling message will also generate two rows of '=' printed. Is this ok? This also applies to all cases within a loop for each uid in the following PR messages.